### PR TITLE
test: Make it easier to use simulated time in integration test, and switch hds_integration_test to use it.

### PR DESCRIPTION
**Pull request by jmarantz**
Description:
 * A previous refactor around TimeSystem lifetime in tests left the previous mechanism partially around in a zombie state. This cleans those up. 
 * There was a problematic pattern in a couple of integration tests of having the test-class inherit from SimulatedTimeSystem. This failed due to a conflicting implementation of timeSystem(). It's resolved by instead inheriting from a class that contains a SimulatedTimeSystem, which has the same effect of establishing -- for the duration of the test method -- the SimulatedTimeSystem as being the current live time-system singleton.
 * There was a performance problem using SimulatedTimeSystem::waitFor with very short durations, which was fixed using a min().
 * Switches hds_simulation_test to SimulatedTimeSystem.
Risk Level: low
Testing: //test/...
Docs Changes: n/a
Release Notes: n/a

### CODE REVIEW

**Comment by jmarantz** [comment]
@sschepens ptal

**Comment by jmarantz** [comment]
Hold off on review -- I thought that echo2_integration_test had been brought in and thus I could change this integration test API in one PR now. I'll restore the realTime() and 3-arg ctor form in a bit.

**Comment by jmarantz** [comment]
Reverted incompatible changes with envoy-filter-example.

**Comment by sschepens** [comment]
@jmarantz seems legit, thanks for taking the time!

File: test/test_common/simulated_time_system.h [hunk 1]
```diff
@@ -118,5 +118,13 @@ class SimulatedTimeSystem : public DelegatingTestTimeSystem<SimulatedTimeSystemH
   }
 };
 
+class TestUsingSimulatedTime {
```
Comment (dnoe) [inline test/test_common/simulated_time_system.h]: Can you add a comment for this? Something along the lines of:

"Inherit from this mixin in a test fixture class to use a SimulatedTimeSystem during the test."

**Comment by dnoe** [comment]
Looks like you will need to revert 
` [899e3a5](/envoyproxy/envoy/pull/6139/commits/899e3a54844f49ce61f1dc77c4a5a2fd9466764e)` and then clean that up in a follow up PR. Sorry for misleading you.

**Pull request merged**